exchanges/gateio/gateio_websocket_request_spot.go

```diff
@@ -261,18 +207,13 @@ func (g *Gateio) SendWebsocketRequest(ctx context.Context, channel string, connS
 		return fmt.Errorf("%s: %s", wsErr.Errors.Label, wsErr.Errors.Message)
 	}
 
-	to := struct {
-		Result any `json:"result"`
-	}{
-		Result: result,
-	}
-
-	return json.Unmarshal(inbound.Data, &to)
+	return json.Unmarshal(inbound.Data, &funnelResult{Result: result})
```

Comment:
Something for another time is that I think jsonparser may be of use in the 
```
if err := json.Unmarshal(endResponse, &inbound); err != nil {
```
above this. All you do is check a status and then if failed, the error message. Might be able to save on a double unmarshal. But Sonic the json hedgehog might be faster I can't benchmark that

Reply:
Yeah good point; I will do some tests on it when I get some time, from what I have seen previously its still lighter to unmarshal everything using sonic rather than even json parser for one item. But mem bad 🤯. Also it might add in overhead from a non error path situation if we look at it from that perspective. We shall see.